source/flux/parts/stoprint.js

```diff
@@ -78,16 +80,52 @@ export function updatePrinters(): ThunkAction<UpdateAllPrintersAction> {
 			return dispatch({type: UPDATE_ALL_PRINTERS_FAILURE, payload: successMsg})
 		}
 
-		const [allPrinters, recentAndPopularPrinters] = await Promise.all([
+		const [
+			allPrintersResponse,
+			recentAndPopularPrintersResponse,
+			colorPrintersResponse,
+		] = await Promise.all([
 			fetchAllPrinters(username),
 			fetchRecentPrinters(username),
+			fetchColorPrinters(),
 		])
 
-		const {recentPrinters, popularPrinters} = recentAndPopularPrinters
+		if (allPrintersResponse.error) {
```

Comment:
This and the following two `if` statements could be condensed to

```js
if(allPrintersResponse.error || recentAndPopularPrintersResponse.error || colorPrintersResponse.error) {
  return dispatch({
    type: UPDATE_ALL_PRINTERS_FAILURE,
    payload: [either a ternary or some kind of fancy select results]
  })
}
```

Otherwise, I'd chain these together as `else if`s, because they will get executed in sequence and if one of them is selected, the chain aborts. (In fact, the function returns.)

Reply:
Since each `if` statement returns if triggered, isn't there no need for `else if`'s? A ternary wouldn't work since there are three conditions right?

Reply:
Well. I suppose.

I would just structure the ternary as:

```
allPrintersResponse.error ? allPrintersResponse.value : 
  (recentAndPopularPrintersResponse.error ? recentAndPopularPrintersResponse.value :
    (colorPrintersResponse.error ? colorPrintersResponse.value : null))

But the more I look at that, the less I like it.

Personally I'd be tempted to write a function that, when given a list of things with `.error` and `.value` values, returns the first one with `value`.  Then use it here.

This way you DRY up the code, which is good. But it's a major gripe of mine.

Reply:
Yeah lol that's why I was afraid of the ternary route.. 😆I tried the following, but it really pissed flow off: 
```javascript
const errorResponse = [allPrintersResponse, recentAndPopularPrintersResponse, colorPrintersResponse].find(response => (
	response.error
))
if (errorResponse && errorResponse.error) {
	return dispatch({
		type: UPDATE_ALL_PRINTERS_FAILURE,
		payload: errorResponse.value
	})
}
```
It can't seem to figure out that if this if condition isn't true, then the value attribute of each response will not be a string, but the proper ResponseType...

Reply:
I'm tempted to leave it as is rather than entering a wrestling match with flow...is that ok @rye ?

Reply:
I'm fine with the way it is. I think it's readable and doesn't have any magic. No extra functions to go look at.